tests/test_audit_logger.py

```
@@ -0,0 +1,47 @@
+import pytest
+from unittest.mock import MagicMock
+
+from lib.audit_logger import is_telegram_ip, log_security_event, EVT_RATE_LIMITED
+
+
+class TestIsTelegramIp:
+
+    def test_accepts_ip_in_first_range(self):
+        # 149.154.160.0/20 covers 149.154.160.0 – 149.154.175.255
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)